File: output/c/hex_to_ansi.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void hex_to_ansi(const char *hex, char *out) {
    unsigned int r, g, b;
    sscanf(hex+1, "%02x%02x%02x", &r, &g, &b);
    sprintf(out, "\033[38;2;%u;%u;%um", r, g, b);
}

int main() {
    char ansi[20];
    hex_to_ansi("#FF4500", ansi);
    printf("%sOrange text\033[0m\n", ansi);
    return 0;
}

File: output/python/hex_to_ansi.py
def hex_to_ansi(hex_code):
    hex_code = hex_code.lstrip('#')
    r = int(hex_code[0:2], 16)
    g = int(hex_code[2:4], 16)
    b = int(hex_code[4:6], 16)
    return f"\033[38;2;{r};{g};{b}m"

if __name__ == "__main__":
    print(hex_to_ansi("#FF4500") + "Orange text" + "\033[0m")
